ddp-workspace/projects/ddp-sdk/src/lib/services/analyticsEvents.service.ts

```diff
@@ -1,78 +1,87 @@
+        this.events.next(this.buildGTagEvent(eventCategory, eventAction));
+    }
 
-  public emitCustomGtagEvent(eventName: string, clickText?: string, clickUrl?: string): void {
-    const newEvent = this.buildGTagEvent(eventName, clickText, clickUrl);
-    this.events.next(newEvent);
-    return;
-  }
+    public emitCustomGtagEvent(eventName: string, clickText?: string, clickUrl?: string): void {
+        const newEvent = this.buildGTagEvent(eventName, clickText, clickUrl);
+        this.events.next(newEvent);
+        return;
+    }
 
-  public emitNavigationEvent(): void {
-    const event: AnalyticsEvent = {
-      hitType: 'pageview',
-      // set page directly in order to exclude sensitive query params
-      // and simpler aggregation of stats
-      page: location.pathname,
-      location: this.location
-    };
-    this.events.next(event);
-    // this is a gtag "recommended event" https://developers.google.com/tag-platform/gtagjs/reference/events#page_view
-    this.events.next(this.buildGTagEvent('page_view'));
-  }
+    public emitNavigationEvent(): void {
+        const event: AnalyticsEvent = {
+            hitType: 'pageview',
+            // set page directly in order to exclude sensitive query params
+            // and simpler aggregation of stats
+            page: location.pathname,
+            location: this.location
+        };
+        this.events.next(event);
+        // this is a gtag "recommended event" https://developers.google.com/tag-platform/gtagjs/reference/events#page_view
+        this.events.next(this.buildGTagEvent('page_view'));
+    }
 
-  public get analyticEvents(): Observable<AnalyticsEvent> {
-    return this.events.pipe(filter(anEvent => isAnalyticsEvent(anEvent))) as Observable<AnalyticsEvent>;
-  }
+    public get analyticEvents(): Observable<AnalyticsEvent> {
+        return this.events.pipe(filter(anEvent => isAnalyticsEvent(anEvent))) as Observable<AnalyticsEvent>;
+    }
 
-  public get gTagEvents(): Observable<GTagEvent> {
-    return this.events.pipe(filter(anEvent => isGtagEvent(anEvent))) as Observable<GTagEvent>;
-  }
+    public get gTagEvents(): Observable<GTagEvent> {
+        return this.events.pipe(filter(anEvent => isGtagEvent(anEvent))) as Observable<GTagEvent>;
+    }
 
-  private get location(): string {
-    return `${location.protocol}//${location.hostname}${location.port ? `:${location.port}` : ``}${location.pathname}`;
-  }
+    /* There does not appear to be a way to prevent gtag.js from automatically sending page view events
+    despite what their documentation says here:
+    https://developers.google.com/analytics/devguides/collection/gtagjs#disable_pageview_measurement
+    Use this option to avoid sending duplicate page_view events given gtag is going to send its own no matter
+    what */
+    public get gTagEventsExcludingPageViews(): Observable<GTagEvent> {
+        return this.gTagEvents.pipe(filter(event => event.event_name !== 'page_view'));
+    }
 
```

Comment:
Please read comment above. Tried as much as I could and could not stop gtag.js from sending `page_view` on its own. Saw some comments in stackoverflow that folks developing for react faced the same thing.
So we are just going to filter those out. For Singular purposes the generic gtag.js event should be acceptable.